packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -380,7 +380,7 @@ List<Error> _validateAst(Root root, String source) {
   final List<Error> result = <Error>[];
   final List<String> customClasses =
       root.classes.map((Class x) => x.name).toList();
-  final List<String> customEnums = root.enums.map((Enum x) => x.name).toList();
+  final Iterable<String> customEnums = root.enums.map((Enum x) => x.name);
```

Comment:
If the only use of this collection is to do tons of `contains` checks on it, why not make it a `Set`?

Reply:
The size of it will usually be really small, less than 10.  For small sizes linear search beats other searches.  For example, here's someone that compared binary search to linear search, https://dirtyhandscoding.wordpress.com/2017/08/25/performance-comparison-linear-search-vs-binary-search/ linear was comparable or faster up to 32 items.

Since we do call it a lot, it might actually be faster to convert it to a list like it was.  The difference wasn't noticeable when running tests.